docs/03-concepts/02-inflight.md

```diff
@@ -49,3 +49,106 @@ Wing is designed around this new programming paradigm, which we call
 **cloud-oriented programming**.
 
 
+
+### Diving deeper into inflight functions
+
+> Taken from [Inflight Magazine](https://docs.winglang.io/blog/2022/12/28/magazine-001#diving-deeper-into-inflight-functions)
+
+[Inflight functions](https://docs.winglang.io/concepts/inflights) are likely Wing's most significant
+language innovation. Since cloud applications are distributed systems and code is executed on
+multiple machines across various compute platforms, our language needs first-class support for this
+idea: code that executes in the cloud and interacts with cloud resources around it. This is where
+inflight functions come into play.
+
+Asynchronous and concurrent programming have been a continuous source for programming language
+innovation and research. Since the early days of
+[`fork()`](https://man7.org/linux/man-pages/man2/fork.2.html) in POSIX, multi-threaded programming,
+[Goroutines](https://gobyexample.com/goroutines), [Erlang
+Processes](https://www.erlang.org/doc/reference_manual/processes.html) and the recently popularized,
+cooperative concurrency with `async/await`, developers have been looking for ways to write
+applications that included more than a single thread of execution in a safe and intuitive manner.
+
+Wing's inflight functions can be thought of as a type of asynchronous function that can be
+**packaged and executed on a remote system**, such as inside a [Kubernetes
+Pod](https://kubernetes.io/docs/concepts/workloads/pods/), [AWS
+Lambda](https://aws.amazon.com/lambda/), [Azure
+Functions](https://azure.microsoft.com/en-us/products/functions/), on [Cloudflare Edge
+Workers](https://workers.cloudflare.com/) or on any other cloud compute platform.
```

Comment:
I wonder if you want to add a sentence like

> While inflights are typically compiled into machine code (like JavaScript), Wing may also be able to compile them into state machines and other formats in the future.

Reply:
I think we should